packages/google_sign_in/google_sign_in_platform_interface/lib/src/method_channel_google_sign_in.dart

```diff
@@ -81,10 +81,10 @@ class MethodChannelGoogleSignIn extends GoogleSignInPlatform {
 
   @override
   Future<List<String>> listMissingScopes(List<String> scopes) {
-    return channel.invokeMethod<List<String>>(
+    return channel.invokeMethod<List<dynamic>>(
       'listMissingScopes',
       <String, List<String>>{'scopes': scopes},
-    );
+    ).then((result) => List<String>.from(result));
```

Comment:
Does this work?

`(result) => result.cast<String>()`;

Reply:
(I wonder why you needed to change from List<String> to List<dynamic>?)

Reply:
I think that would work, Effective Dart says to prefer List.from over cast: https://dart.dev/guides/language/effective-dart/usage#dont-use-cast-when-a-nearby-operation-will-do

invokeMethod apparently doesn't support generic type parameters other than dynamic? With the initial approach, the future never completed. This change seemed to resolve the issue I was seeing in manual testing, but I've never seen dart act this way before.
https://api.flutter.dev/flutter/services/MethodChannel/invokeMethod.html

Reply:
So weird, I've seen this done in other places, but I didn't understand why. It might be a bug in the MethodChannel :/